src/middlewared/middlewared/plugins/smb.py

```diff
@@ -299,19 +258,17 @@ def check_codec(encoding):
         return True
 
     @private
-    def groupmap_list(self):
-        groupmap_list = []
-        passdb = samba3.Samba3('/usr/local/etc/smb4.conf').get_sam_db()
-        groupmaps = passdb.enum_group_mapping()
-        for g in groupmaps:
-            groupmap_list.append({
-                'comment': g.comment,
-                'gid': g.gid,
-                'ntgroup': g.nt_name,
-                'SID': str(g.sid),
-                'sid_type': LSA_sidType(g.sid_name_use).name,
-            })
-        return groupmap_list
+    async def groupmap_list(self):
+        groupmap = []
+        out = await run([SMBCmd.NET.value, 'groupmap', 'list'], check=False)
+        if out.returncode != 0:
+            raise CallError(f'groupmap list failed with error {out.stderr.decode()}')
+        for line in (out.stdout.decode()).splitlines():
+            m = re.match(r'^(?P<ntgroup>.+) \((?P<SID>S-[0-9\-]+)\) -> (?P<unixgroup>.+)$', line)
```

Comment:
Just a minor nit but may be worth doing, we could make the regex a global var and compile it, so its not compiled every iteration and everytime this method is called

RE_NETGROUPMAP = re.compile(r'...')
or something, then RE_NTHASH.match(line)